crates/ruff_linter/src/rules/pycodestyle/rules/lambda_assignment.rs

```diff
@@ -267,14 +267,21 @@ fn replace_trailing_ellipsis_with_original_expr(
     )
     .unwrap_or(lambda.body.range());
 
-    let original_expr_in_source = checker.locator().slice(original_expr_range);
+    // This prevents the autofix of introducing a syntax error if the lambda's body is an `if`
+    // expression spanning across multiple lines. To avoid the syntax error we wrap the expression
+    // in parenthesis.
+    let original_expr_in_source = if matches!(lambda.body.as_ref(), Expr::If(_)) {
```

Comment:
I don't think this is specific to if expressions. For example, the input could be


```py
foo_tooltip = (
    lambda x, data: f"\nfoo: {data['foo'][int(x)]}" + 
    more

)
```

I think what we should do instead is to test if the expression is parenthesized (see `is_expression_parenthesized`) and if so, preserve the parentheses around the body.